default_gateway.go

```diff
@@ -103,6 +103,9 @@ func (sb *sandbox) needDefaultGW() bool {
 		if ep.getNetwork().Type() == "null" || ep.getNetwork().Type() == "host" {
 			continue
 		}
+		if ep.joinInfo.disableGatewayService {
+			return false
+		}
```

Comment:
`disableGatewayService` is more like a sandbox property. If you enable such service during joining one of its endpoints, others will be infected. Since `disableGatewayService` is false by default and it becomes true if drivers explicitly setting it to true, I suggest we `return false` here.